velero-plugins/serviceaccount/backup.go

```diff
@@ -100,8 +107,8 @@ func (p *BackupPlugin) UpdateSCCMap() error {
 				if namespace == "" {
 					continue
 				}
-				if p.SCCMap[namespace] == nil {
-					p.SCCMap[namespace] = make(map[string][]apisecurity.SecurityContextConstraints)
+				if c.SCCMap[namespace] == nil {
+					c.SCCMap[namespace] = make(map[string][]apisecurity.SecurityContextConstraints)
```

Comment:
Do we need eventual deletion of items from the map somewhere? I expect this map will grow indefinitely ATM?

Reply:
@kaovilai Actually each time we repopulate this we get the full list of SCCs in-cluster, and then populate all SAs for all namespaces which map to it. So the size should only grow as the number of SCCs and related SAs grows. We update it once for a backup to make sure it's up-to-date as of backup start time. Realistically, we probably don't need to track by backup since the plugin currently exits after backup processing, so next backup starts with an empty map anyway, but verifying per backup makes the plugin resiliant to that design choice.

tl;dr this map shouldn't grow -- it should always be the size of the actual in-cluster mapping between all SAs and all SCCs.

Reply:
I guess I need to look at how plugin exits.. and if it kills/clears all memory usage.

Reply:
@kaovilai There's a cleanup call made at the end of the the reconcile loop that kills the plugin process. We had a Velero bug a few months ago where in some cases this wasn't happening properly, and plugin processes piled up (that's fixed now, though). Easy enough to confirm by watching processes in the pod while running a backup. If the process exits, surely that means it's no longer consuming memory.